app/scripts/utils/clone-event.js

```diff
@@ -1,10 +1,17 @@
+// @ts-check
+
 /**
- * Clone an event by invoking the source event's constructor and passing in
- *   the source event.
- * @param   {object}  event  Source event to be cloned.
- * @return  {object}  Cloned event
+ * Clone an event by invoking the source event's constructor and passing in the source event.
+ *
+ * @template {Event & { sourceUid?: string, forwarded?: boolean }} TEvent
+ * @param {TEvent} event - Source event to be cloned.
+ * @return {TEvent} Cloned event
  */
 const cloneEvent = (event) => {
+  if (!(event instanceof Event)) {
```

Comment:
This type of cloning only works if the `event` is really an event (not just the "event" interface)